site/content/extend/plugins/signing-plugins.md

```diff
@@ -0,0 +1,151 @@
+---
+title: "Signing Plugins"
+date: 2019-09-04T00:00:00-05:00
+subsection: Plugins (Beta)
+weight: 50
+---
+
+"Plugin Marketplace” is a platform used to distribute common plugins where Mattermost admins can install plugins without verification. To trust a plugin, there needs to be a way to verify the authenticity and integrity of a plugin before being installed. [The GNU Privacy Guard (GPG)](https://gnupg.org) tool is used to sign and verify plugins. Every plugin is signed using a private key and every Mattermost instance should have access to a public key which is used to verify plugin signatures on installation.
+
+## Key Generation
+Public and private key pairs are needed to sign and verify plugins. Private key is used for signing and should be kept in the Vault. Public key is used for verification and can be distributed freely. To generate a key pair, run the following command:
+
+
+ ```console
+ > gpg --full-generate-key
+
+ Please select what kind of key you want:
+    (1) RSA and RSA (default)
+    (2) DSA and Elgamal
+    (3) DSA (sign only)
+    (4) RSA (sign only)
+ Your selection? 1
+
+ RSA keys may be between 1024 and 4096 bits long.
+ What keysize do you want? (2048) 3072
+
+ Requested keysize is 3072 bits
+
+ Please specify how long the key should be valid.
+          0 = key does not expire
+       <n>  = key expires in n days
+       <n>w = key expires in n weeks
+       <n>m = key expires in n months
+       <n>y = key expires in n years
+ Key is valid for? (0) 0
+
+ Key expires at ...
+
+ Is this correct? (y/N) y
+
+ GnuPG needs to construct a user ID to identify your key.
+ Real name: Mattermost Inc
+
+ Email address: info@mattermost.com
+
+ Comment:
+
+ You selected this USER-ID:
+     "Mattermost Inc <info@mattermost.com>"
+ Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O
+
+ ```
+ Note: key size should be at least 3072 bits.
+
+### Exporting the Private Key
+Find the ID of your private key first. The ID is a hexadecimal number.
+```console
+> gpg --list-secret-keys
+```
+This is your private key and you should keep it secret. Replace `XXXXXXXX` with your hexadecimal key ID.
```

Comment:
It seems we have to repeat the instructions on `XXXXXXXX` multiple times in this document. I'd suggest actually just running the steps, showing the output, and using the value in that output in all the examples below, with a single line around here stating that your keys will, of course, be different.